feedback/extensions/filters.py

```diff
@@ -0,0 +1,153 @@
+            "section_key": BLOCK_CATEGORY,
+            "section_display_name": "Course Feedback",
+            "course_id": str(course.id),
+            "template_path_prefix": TEMPLATE_ABSOLUTE_PATH,
+        }
+        context["sections"].append(section_data)
+
+        return {"context": context}
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string("feedback", path)
+        return data.decode("utf8")
+
+
+def load_blocks(request, course):
+    """
+    Load feedback blocks for a given course for all enrolled students.
+
+    Arguments:
+        request (HttpRequest): Django request object.
+        course (CourseLocator): Course locator object.
+    """
+    course_id = str(course.id)
+
+    feedback_blocks = modulestore().get_items(
+        course.id, qualifiers={"category": "feedback"}
+    )
+
+    blocks = []
+    students = CourseEnrollment.objects.filter(course_id=course_id).values_list(
+        "user_id", "user__username"
+    )
+    for feedback_block in feedback_blocks:
+        block, _ = get_block_by_usage_id(
+            request,
+            str(course.id),
+            str(feedback_block.location),
+            disable_staff_debug_info=True,
+            course=course,
+        )
+        answers = load_xblock_answers(
+            request,
+            students,
+            str(course.location.course_key),
+            str(feedback_block.location),
+            course,
+        )
+
+        vote_aggregate = []
+
+        if not block.vote_aggregate:
+            block.vote_aggregate = [0] * len(block.get_prompt()["scale_text"])
+        for i, vote in enumerate(block.vote_aggregate):
+            vote_aggregate.append(
+                {
+                    "scale_text": block.get_prompt()["scale_text"][i],
+                    "count": vote,
+                }
+            )
+
+        blocks.append(
+            {
+                "display_name": block.display_name,
+                "prompts": block.prompts,
+                "vote_aggregate": vote_aggregate,
+                "answers": answers,
+            }
+        )
+    return blocks
+
+
+def load_xblock_answers(request, students, course_id, block_id, course):
+    """
+    Load answers for a given feedback xblock instance.
+
+    Arguments:
+        request (HttpRequest): Django request object.
+        students (list): List of enrolled students.
+        course_id (str): Course ID.
+        block_id (str): Block ID.
+        course (CourseDescriptor): Course descriptor.
+    """
+    answers = []
+    for user_id, username in students:
+        student_xblock_instance = load_single_xblock(
+            request, user_id, course_id, block_id, course
+        )
+        if student_xblock_instance:
+            prompt = student_xblock_instance.get_prompt()
+            if student_xblock_instance.user_vote == -1 and not student_xblock_instance.user_freeform:
```

Comment:
what's the `-1` for?

Reply:
It means the user didn't vote. If the user didn't vote and there is no feedback we don't add it to the template